frontend/ui/widget/filechooser.lua

```diff
@@ -241,6 +241,12 @@ local FileChooser = Menu:extend{
                         return
                     end
 
+                    -- books marked as "on hold" should be considered the same as 0%
+                    if summary and summary.status == "abandoned" then
+                        item.percent_finished = 0.0
```

Comment:
At a glance this looks very suspicious because it's acting straight on `item`. (Which is to say, won't that have undesired side effects?)

Reply:
I am strictly following the existing code base, so if using `item` in that case is not a good practice, it would be a good idea to change the previous instance (line 240) that also utilises `item` as well. Await further instructions.

Having said that, I am running that very code on a kindle 5 (non touch) and, it is working wonderfully. I am able to label books as "on hold" and they sort themselves out just fine, I changed pages while on hold and it continues to behave properly, also, switching back to "reading" sends them back to their respective positions in the percentage queue. So it seems everything is running smoothly.

What exactly did you have in mind regarding side effects? I could try recreating some if you have any specific ideas but again, so far so good.

Reply:
That is why the function name is `item_func`.

Reply:
> I am strictly following the existing code base, so if using `item` in that case is not a good practice, it would be a good idea to change the previous instance (line 240) that also utilises `item` as well. Await further instructions.

I'm just asking you who looked at the code what item is in this case. :-) But yes, the same question would indeed apply a few lines up.

A side effect would mean showing (or using) the wrong percentage somewhere outside of sorting order. The fact that it's currently exclusive between percentage and hold/finished (I think) is orthogonal to that, but in any case that specific scenario is fine.

@hius07 Do you mean everything's safe and sound?

Reply:
I mean that originally `item_func` was designed to act straight on `item` (a table) being passed as arg, it was okay.

It is a very good point: `percent_finished` is shown in mandatory, so it will display the wrong value for finished and "on hold" books in this new collate type.

Reply:
so what would be the best way to do it?

Reply:
My instinct would be to simply add a special `sorting_percentage` (not necessarily by that name, but by that principle). 

As for this PR it's probably fine if you don't change anything about it, but let's await what @hius07 says since he knows this part of the code best at this point.

Reply:
Got it. I  tidied up earlier, but no real changes, just moving everything into one singe `if` statement. 

Holding down for now then.